test/serializers/cache_test.rb

```diff
@@ -64,15 +66,21 @@ def test_cache_key_definition
         assert_equal(nil, @comment_serializer.class._cache_key)
       end
 
-      def test_cache_key_interpolation_with_updated_at
+      def test_error_is_raised_if_cache_key_is_not_defined_on_object_or_passed_as_cache_option
+        assert_raises ActiveModel::Serializer::Adapter::CachedSerializer::UndefinedCacheKey do
+          render_object_with_cache(@article)
+        end
+      end
+
+      def test_cache_key_interpolation_with_updated_at_when_cache_key_is_not_defined_on_object
         render_object_with_cache(@author)
-        assert_equal(nil, ActionController::Base.cache_store.fetch(@author.cache_key))
-        assert_equal(@author_serializer.attributes.to_json, ActionController::Base.cache_store.fetch("#{@author_serializer.class._cache_key}/#{@author_serializer.object.id}-#{@author_serializer.object.updated_at.strftime("%Y%m%d%H%M%S%9N")}").to_json)
+        key = cache_key_with_adapter("#{@author_serializer.class._cache_key}/#{@author_serializer.object.id}-#{@author_serializer.object.updated_at.strftime("%Y%m%d%H%M%S%9N")}")
+        assert_equal(@author_serializer.attributes.to_json, ActionController::Base.cache_store.fetch(key).to_json)
```

Comment:
- [ ] TODO should use `ActiveModelSerializers.config.cache_store` instead, but maybe in another PR